Sources/LanguageServerProtocol/SupportTypes/MacroExpansion.swift

```diff
@@ -0,0 +1,25 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// The expansion of a macro in a source file.
+public struct MacroExpansion: ResponseType, Hashable {
+  /// The position in the source file where the expansion would be inserted.
+  public let position: Position
+
+  /// The Swift code that the macro expands to.
+  public let sourceText: String
```

Comment:
I would like to include the buffer name here as well. My thinking is that we could to encode it like

```
swift-macro-expansion:///path/to/file.swift?buffer=@swiftmacro_...
```

That way the editor could open the buffer with that URI and we could start supporting semantic functionality (like nested expansion) inside the expansion buffer as well.

Reply:
Yes, I like the idea (I'll also refer to my longer comment here)